### PR TITLE
Various plotting fixes

### PR DESCRIPTION
**Pull request by petersteneteg**

### CODE REVIEW

File: modules/plotting/processors/dataframeexporter.cpp [hunk 1]
```diff
@@ -56,7 +57,8 @@ DataFrameExporter::DataFrameExporter()
     , exportFile_("exportFile", "Export file name", "", "dataframe")
     , exportButton_("snapshot", "Export DataFrame")
     , overwrite_("overwrite", "Overwrite", false)
-    , forceDoublePrecision_("forceDoublePrecision", "Force Double Precision", true)
+    , separateVectorTypesIntoColumns_("separateVectorTypesIntoColumns",
+                                      "Separate Vector Types Into Columns", true)
```
Comment (martinfalk) [inline modules/plotting/processors/dataframeexporter.cpp:61]: 👍

File: modules/plotting/processors/dataframeexporter.cpp [hunk 2]
```diff
@@ -84,67 +86,83 @@ void DataFrameExporter::exportNow() {
...
+    const char lineterminator = '\n';
+    const std::array<char, 4> componentNames = {'X', 'Y', 'Z', 'W'};
 
     // headers
-    for (size_t j = 0; j < dataFrame->getNumberOfColumns(); j++) {
-        auto c = dataFrame->getColumn(j);
-        auto components = c->getBuffer()->getDataFormat()->getComponents();
-
-        if (components > 1 && forceDoublePrecision) {
+    auto oj = util::make_ostream_joiner(file, delimiter);
```
Comment (martinfalk) [inline modules/plotting/processors/dataframeexporter.cpp:111]: won't a ostream_joiner add a delimiter after the last element as well?

File: modules/plotting/processors/dataframeexporter.cpp [hunk 3]
```diff
@@ -84,67 +86,83 @@ void DataFrameExporter::exportNow() {
...
+        auto df = col->getBuffer()->getDataFormat();
+        if (auto cc = dynamic_cast<const CategoricalColumn*>(col.get())) {
+            printers.push_back([cc](std::ostream& os, size_t index) {
+                os << "\"" << cc->getAsString(index) << "\"";
+            });
+        } else if (df->getComponents() == 1) {
+            col->getBuffer()
+                ->getRepresentation<BufferRAM>()
+                ->dispatch<void, dispatching::filter::Scalars>([&printers](auto br) {
+                    using ValueType = util::PrecsionValueType<decltype(br)>;
```
Comment (martinfalk) [inline modules/plotting/processors/dataframeexporter.cpp]: ValueType is unused

File: modules/plotting/processors/dataframeexporter.cpp [hunk 4]
```diff
@@ -84,67 +86,83 @@ void DataFrameExporter::exportNow() {
...
+                        for (size_t i = 0; i < util::flat_extent<ValueType>::value; ++i) {
+                            oj = br->getDataContainer()[index][i];
+                        }
+                    });
+                });
+        } else {
+            col->getBuffer()
+                ->getRepresentation<BufferRAM>()
+                ->dispatch<void, dispatching::filter::Vecs>([&printers](auto br) {
+                    using ValueType = util::PrecsionValueType<decltype(br)>;
```
Comment (martinfalk) [inline modules/plotting/processors/dataframeexporter.cpp]: ValueType is unused

File: modules/plotting/processors/imagetodataframe.cpp [hunk 5]
```diff
@@ -50,82 +52,199 @@ const ProcessorInfo ImageToDataFrame::processorInfo_{
...
+                    &col->getTypedBuffer()->getEditableRAMRepresentation()->getDataContainer());
+            }
+
+            auto& magnitudes = dataFrame->addColumn<float>("Magnitude", size)
+                                   ->getTypedBuffer()
+                                   ->getEditableRAMRepresentation()
+                                   ->getDataContainer();
+
+            std::vector<float>* greycalePerceived = nullptr;
+            std::vector<float>* greycaleRelative = nullptr;
```
Comment (martinfalk) [inline modules/plotting/processors/imagetodataframe.cpp]: typos `grey_caleRelative` and `grey_calePerceived`

**Review by martinfalk — APPROVED** [review]
nice additions! Only minor comments

File: modules/plotting/processors/dataframeexporter.cpp (hunk 2 above)
Comment (petersteneteg) [reply modules/plotting/processors/dataframeexporter.cpp:111]: mine won't ;)

File: modules/plotting/processors/imagetodataframe.cpp (hunk 5 above)
Comment (martinfalk) [reply modules/plotting/processors/imagetodataframe.cpp]: -> `gray` if we stick to American English ;-)

**Review by martinfalk — APPROVED** [review]

**Pull request merged**